[React] Fix accepted types for some hooks

atharvadeosthale opened the pull request:
Hooks `useUnclaimedNFTs()`, `useClaimedNFTs()`, `useUnclaimedNFTSupply()`, and `useClaimedNFTSupply()` don't work with edition drop contracts, however, the comments on these hooks specified they could.

nachoiacovino reviewed (changes requested):
we can't accept only NFTDrop which is the prebuilt, we need to accept DropContract (that has all drops) but exclude all erc1155 and erc20, so find how we do it elsewhere, but need to do a DropContract exclude EditionDrop, TokenDrop

atharvadeosthale commented:
`useClaimedNFTs` and `useUnclaimedNFTs` took only `NFTDrop` even before so I changed their link to point to `NFTDrop`. For the ones where we exclude edition drop and token drop I've added normal text and removed links

nachoiacovino commented on packages/react-core/src/evm/hooks/async/drop.ts:
````diff
@@ -74,10 +76,10 @@ export function useUnclaimedNFTs(
  * ```
  *
- * @param contract - an instance of a {@link DropContract}
+ * @param contract - an instance of a {@link NFTDrop}
````
this is not correct now

nachoiacovino reviewed (changes requested):
fix the params plz :D

nachoiacovino reviewed (approved):
lgtm!
